services/QuillLMS/app/workers/skills_report/cache_users_api_data_worker.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+module SkillsReport
+  class CacheUsersApiDataWorker
+    include Sidekiq::Worker
+
+    sidekiq_options queue: SidekiqQueue::CRITICAL
+
+    attr_reader :user_id, :classroom_id
+
+    def perform(user_id, classroom_id)
+      @user_id = user_id
+      @classroom_id = classroom_id
+      return unless user
+      return unless classroom
+
+      concept_ids.each do |concept_id|
+        all_history_options.each do |all_history|
+          ell_options.each do |ell|
+            cache_data(all_rows, classroom:, concept_id:, all_history:, ell:)
+          end
+        end
+      end
+    end
+
+    private def cache_data(rows, classroom:, concept_id:, all_history:, ell:)
+      filtered_rows = filter_rows_by_params(rows, concept_id:, all_history:, ell:)
+
+      user.classroom_cache(classroom,
+        key: SkillsReportsController::PERFORMANCE_BREAKDOWN_BY_STUDENT_CACHE_KEY,
+        groups: {
+          classroom_id:,
+          concept_id:,
+          all_history:,
+          ell:
+        }) do
+        ComposeScoresByUser.run(question_responses: filtered_rows, questions_hash:, user_hash:)
+      end
+    end
+
+    private def all_history_options = [false, true]
+    private def ell_options = [false, true]
+    private def all_rows = @all_rows ||= QuestionResponsesForClassroom.run(classroom_id:, concept_id: nil, for_cache: true)
+    private def concept_ids = concepts.map(&:id)
+    private def concepts = @concepts ||= Concept.in_skills_report.visible.select(:id, :name, :parent_id, :ell, :gen_ed)
+    private def question_uids = all_rows.map(&:question_uid).uniq
+    private def questions_hash = @questions_hash ||= Question.where(uid: question_uids).group_by(&:uid)
+    private def user = @user ||= User.find_by(id: user_id)
+    private def user_hash = @user_hash ||= User.where(id: user_ids).pluck(:id, :name).to_h
+    private def user_ids = all_rows.map(&:user_id).uniq
+
+    private def filter_classroom_history(rows, classroom_id) = rows.filter { |row| row.classroom_id == classroom_id }
```

Comment:
This method and several others are duplicates of those in `cache_skills_api_data_worker`

Reply:
I think that this is a good call-out, but will address DRY issues in a separate PR to keep this simple.